lib/daru/dataframe.rb

```diff
@@ -2015,6 +2015,22 @@ def split_by_category cat_name
       end
     end
 
+    # returns array of row tuples at given index(s)
+    def access_row_tuples_by_indexs *indexes
+      positions = @index.pos(*indexes)
+
+      return populate_row_for(positions) if positions.is_a? Numeric
+
+      res = []
+      new_rows = @data.map { |vec| vec[*indexes] }
+      indexes.each do |index|
+        tuples = []
+        new_rows.map { |row| tuples += [row[index]] }
+        res << tuples
+      end
+      res
+    end
```

Comment:
Why do you need this method here? If it is necessary only to support algorithms in "group by", maybe it is better to be there?.. Please provide an explanation for this method, it is hard to undestand from docs as well as from code.